app/src/main/java/unipd/se18/ocrcamera/TextExtractor.java

```diff
@@ -0,0 +1,240 @@
+package unipd.se18.ocrcamera;
+
+import android.graphics.Bitmap;
+import android.util.Log;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.ml.vision.FirebaseVision;
+import com.google.firebase.ml.vision.common.FirebaseVisionImage;
+import com.google.firebase.ml.vision.text.FirebaseVisionText;
+import com.google.firebase.ml.vision.text.FirebaseVisionTextRecognizer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Class that implements the common OCR interface for retrieving text from a Bitmap image.
+ * This class uses ml-kit provided by Firebase.
+ * More details at: {@link FirebaseVisionText}.
+ * @see <a href="https://firebase.google.com/docs/ml-kit/android/recognize-text">
+ *     ml-kit, recognize text</a>
+ * @author Pietro Prandini (g2)
+ */
+public class TextExtractor implements OCRInterface {
+    /**
+     * String used for the logs of this class.
+     */
+    private final String TAG = "TextExtractor";
+
+    /*
+    The next four int are used for recognizing the position of the FirebaseVisionText Objects.
+    They could be useful for sorting the blocks or for an automatic recognition
+    of the ingredients text block.
+     */
+
+    /**
+     * Index of the point on the top-left position.
+     * More details at:
+     * {@link FirebaseVisionText.TextBlock#getCornerPoints()},
+     * {@link FirebaseVisionText.Line#getCornerPoints()}
+     * or {@link FirebaseVisionText.Element#getCornerPoints()}.
+     */
+    private final int TOP_LEFT = 0;
+
+    /**
+     * Index of the point on the top-right position.
+     * More details at:
+     * {@link FirebaseVisionText.TextBlock#getCornerPoints()},
+     * {@link FirebaseVisionText.Line#getCornerPoints()}
+     * or {@link FirebaseVisionText.Element#getCornerPoints()}.
+     */
+    private final int TOP_RIGHT = 1;
```

Comment:
I suggest to delete these variables that are not used and relative comments for a better clean code

Reply:
Thanks, I've left these variables for the near future about sorting blocks or recognizing the block of the ingredients. Could I remove eventually some comments?

Reply:
Yes, maybe you can group the comments into one like
> These variables are used for sorting the blocks. 

And if needed you can add more comments in the future methods

Reply:
Ok, thanks, I'll push a commit with the requested changes soon.